app/scripts/views/components/barChart/barChart.js

```diff
@@ -15,61 +15,86 @@
     var initDc_ = function(logScale) {
       var tickVal = [];
       var i = 0;
+      var barSet = new Set();
+
+        dcDimension = ndx_.dimension(function (d) {
+            var val = d[data_.attrId];
+            var _min;
+            var _max;
+
+             //isNaN(val) treats string in data as 'NA', such as "withheld" and "cannotReleaseHIPAA"
+            if (iViz.util.strIsNa(val, true) || isNaN(val)) {
+                val = opts_.xDomain[opts_.xDomain.length - 1];
+            } else if (logScale) {
+                for (i = 1; i < opts_.xDomain.length; i++) {
+                    if (d[data_.attrId] < opts_.xDomain[i] &&
+                        d[data_.attrId] >= opts_.xDomain[i - 1]) {
+                        val = parseInt(Math.pow(10, i / 2 - 0.25), 10);
+                        _min = opts_.xDomain[i - 1];
+                        _max = opts_.xDomain[i];
+                        break;
+                    }
+                }
+            } else if (data_.smallDataFlag) {
+                if (d[data_.attrId] <= opts_.xDomain[1]) {
+                    val = opts_.xDomain[0];
+                } else if (d[data_.attrId] > opts_.xDomain[opts_.xDomain.length - 3]) {
+                    val = opts_.xDomain[opts_.xDomain.length - 2];
+                } else {
+                    for (i = 2; i < opts_.xDomain.length - 2; i++) {
+                        if (d[data_.attrId] <= opts_.xDomain[i] &&
+                            d[data_.attrId] >= opts_.xDomain[i - 1]) {
+                            _min = opts_.xDomain[i - 1];
+                            _max = opts_.xDomain[i];
+                            val = _min + (_max - _min) / 3;
+                            break;
+                        }
+                    }
+                }
+            } else if (!data_.valueAsTick) {
```

Comment:
What happened if data.smaillDataFlag is true, and data_.valueAsTick is true as well?